lightning/src/ln/peer_handler.rs

```diff
@@ -1036,8 +1036,9 @@ impl<Descriptor: SocketDescriptor, CM: Deref> PeerManager<Descriptor, CM> where
 
 	/// Indicates that the given socket descriptor's connection is now closed.
 	///
-	/// This must be called even if a PeerHandleError was given for a read_event or write_event,
-	/// but must NOT be called if a PeerHandleError was provided out of a new_\*\_connection event!
+	/// This must only be called if the socket has been lost and must NOT be called in any case
+	/// where other parts of this library (eg PeerHandleError, explicit disconnect_socket calls)
+	/// instruct you to disconnect the peer.
 	///
 	/// Panics if the descriptor was not previously registered in a successful new_*_connection event.
 	pub fn disconnect_event(&self, descriptor: &Descriptor) {
```

Comment:
Why not rename disconnect_event_connection_lost or disconnect_event_socket_lost, by the way you may also comment what the function do on the internal state, not only how and when you should call it